icu4c/source/i18n/unum.cpp

```diff
@@ -112,10 +113,6 @@ unum_open(  UNumberFormatStyle    style,
     case UNUM_DURATION:
         retVal = new RuleBasedNumberFormat(URBNF_DURATION, Locale(locale), *status);
         break;
-
-    case UNUM_NUMBERING_SYSTEM:
-        retVal = new RuleBasedNumberFormat(URBNF_NUMBERING_SYSTEM, Locale(locale), *status);
```

Comment:
Suggestion: I think it would ideally be better to "fix" this in the RuleBasedNumberFormat constructor itself if possible. It seems like it could be disruptive if `unum_open` started returning something that might be a DecimalFormat instead of a RuleBasedNumberFormat.

Reply:
As I thought I explained in the JIRA ticket, I don't think there's a good way to do that.  There are two fundamental problems: 1) A decent number of numbering systems are not implemented with a `RuleBasedNumberFormat`, and 2) there's no way to set the default rule set of an RBNF through the C interface.

So if you ask the RBNF constructor for a numbering system that's not RBNF-backed, you have two choices: The constructor can fail, which seems unnecessarily user-hostile, or you can get a rule set that consists of `0: =#,##0=;`, which is kludgey and may or may not produce the right thing (I put that in anyway, but I'm still not sure I think it's a good idea).

If we leave the RBNF constructor as it is and rely on the caller calling `setDefaultRuleSet()`, the caller has to know the name of the rule set.  I don't believe they're documented or guaranteed to be stable right now, and I'm not sure we want to change that.

It seems safer to say that you have to go through `NumberFormat::getInstance()` or `unum_open()` if you want to pick up the numbering system from the locale ID, and to leave the RBNF constructor as an implementation detail (in fact, if you're working in C, you have no choice).  That leaves us with the question of what the difference is between `UNUM_DECIMAL` and `UNUM_NUMBERING_SYSTEM`.  I agree it's weird to have them do the same thing, but I don't know what the original intent was or whether it really holds up at this point.  Frankly, it's equally weird to have `UNUM_DECIMAL` sometimes return something that's not a `DecimalFormat`.

>So while I think we should fix the odd bug...

Not sure what you meant by "the odd bug" here.  The fact that you get Tamil if you don't specify something else?

>...the high-order bit for me is that we provide proper documentation, e.g., pointing people toward rbnf.setDefaultRuleSet("%ethiopic").

I thought I did that.  There are additional places in the docs where I need to add verbiage?

Reply:
OK, I'm convinced.

> I agree it's weird to have them do the same thing, but I don't know what the original intent was or whether it really holds up at this point.

OK.

> I thought I did that. There are additional places in the docs where I need to add verbiage?

Actually your docs look good and I didn't see all of them on my first pass.

Reply:
Thank you!  That's a relief.